src/k8s/pkg/k8sd/api/endpoints.go

```diff
@@ -89,5 +89,16 @@ func (e *Endpoints) Endpoints() []rest.Endpoint {
 			Path: "kubernetes/auth/webhook",
 			Post: rest.EndpointAction{Handler: e.postKubernetesAuthWebhook, AllowUntrusted: true},
 		},
+		// ClusterAPI management endpoints.
+		{
+			Name: "GenerateJoinToken",
+			Path: "x/capi/generate-join-token",
+			Post: rest.EndpointAction{Handler: e.postClusterJoinTokens, AccessHandler: ValidateCAPIAuthTokenAccessHandler("token")},
```

Comment:
Just for my own understanding: We probably need to update the k8sd-proxy later to only forward requests that target the `x/capi/...` endpoints, right?

Should we also give the token a slightly more verbose name, e.g. `capi-auth-token`?

Reply:
Sounds good! I've changed the token header name